src/gc-common.h

```diff
@@ -185,4 +185,13 @@ extern jl_ptls_t* gc_all_tls_states;
 
 extern int gc_logging_enabled;
 
+// =========================================================================== //
+// Misc
+// =========================================================================== //
+
+// Allocates a new weak-reference, assigns its value and increments Julia allocation
+// counters. If thread-local allocators are used, then this function should allocate in the
+// thread-local allocator of the current thread.
+JL_DLLEXPORT struct _jl_weakref_t *jl_gc_new_weakref(struct _jl_value_t *value);
```

Comment:
Minor style suggestion: since this is in `gc-common.h`, we should have access to `jl_weakref_t` and `jl_value_t`.

Perhaps use those, instead of `struct _jl_weakref_t` and `struct _jl_value_t`?

Reply:
```suggestion
JL_DLLEXPORT jl_weakref_t *jl_gc_new_weakref(jl_value_t *value);
```
Yeah, should work. @fingolfin do you agree?